expression/aggregation/base_func.go

```diff
@@ -286,7 +286,7 @@ func (a *baseFuncDesc) typeInfer4MaxMin(ctx sessionctx.Context) {
 		a.Args[0] = expression.BuildCastFunction(ctx, a.Args[0], tp)
 	}
 	a.RetTp = a.Args[0].GetType()
-	if (a.Name == ast.AggFuncMax || a.Name == ast.AggFuncMin) && a.RetTp.GetType() != mysql.TypeBit {
+	if (a.Name == ast.AggFuncMax || a.Name == ast.AggFuncMin || a.Name == ast.AggFuncFirstRow || a.Name == ast.AggFuncSum || a.Name == ast.AggFuncAvg) && a.RetTp.GetType() != mysql.TypeBit {
```

Comment:
the issue only related to 'a.Name == ast.AggFuncFirstRow', but there may be bugs for "sum" or 'avg'.